### PR TITLE
Add a DR overview focused on resiliency with comparison for HA & DR

### PR DESCRIPTION
**Pull request by kathancox**
Fixes DOC-9928, DOC-9929

This PR (in draft) adds a DR overview page to direct users toward establishing resiliency in their deployments. Currently included this as an overview page for DR page, but there are other options.

[Rendered preview](https://deploy-preview-18490--cockroachdb-docs.netlify.app/docs/v24.1/disaster-recovery-overview.html)

### CODE REVIEW

File: src/current/v24.1/disaster-recovery-overview.md [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+
+{{site.data.alerts.callout_success}}
+For a practical guide on how CockroachDB replicates, distributes, and rebalances data, refer to the [Replication and Rebalancing]({% link {{ page.version.version }}/demo-replication-and-rebalancing.md %}) demo.
+{{site.data.alerts.end}}
+
+## Resilience strategy
+
+As you evaluate CockroachDB's disaster recovery features, consider your organization's requirements for the amount of tolerable data loss and the acceptable length of time to recover.
+
+- Recovery Point Objective (RPO): The maximum amount of time that an organization can tolerate losing data.
```
Comment (alicia-l2) [inline src/current/v24.1/disaster-recovery-overview.md:22]: "maximum amount of data loss – as measured by time – that an organization can tolerate." Maybe this?

File: src/current/v24.1/disaster-recovery-overview.md [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+- [Choose a disaster recovery strategy](#choose-a-disaster-recovery-strategy)
+
+### Choose a high availability strategy
+
+CockroachDB uses synchronous, built-in replication to create and distribute copies of data, ensuring consistency across the data copies. The database can tolerate nodes going offline without service interruption whether in a single-region or [multi-region]({% link {{ page.version.version }}/multiregion-overview.md %}) cluster.
+
+<table class="comparison-chart">
+  <tr>
+    <th></th>
+    <th>Single-region replication</th>
```
Comment (alicia-l2) [inline src/current/v24.1/disaster-recovery-overview.md:41]: Is it worth adding "synchronous" replication here?

File: src/current/v24.1/disaster-recovery-overview.md [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+    <td>1</td>
+    <td>3</td>
+  </tr>
+
+  <tr>
+    <td class="comparison-chart__feature">
+      <b>Fault tolerance</b>
+    </td>
+    <td>Zero RPO node, availability zone failures</td>
+    <td>Zero RPO node, availability zone failures</td>
```
Comment (alicia-l2) [inline src/current/v24.1/disaster-recovery-overview.md:98]: The multi-region one  should also be able to survive a region failure

File: src/current/v24.1/disaster-recovery-overview.md [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+    <td>1</td>
+    <td>2</td>
+  </tr>
+
+  <tr>
+    <td class="comparison-chart__feature">
+      <b>Fault tolerance</b>
+    </td>
+    <td>Not applicable</td>
+    <td>Zero RPO node, availability zone region failure with loss up to RPO</td>
```
Comment (alicia-l2) [inline src/current/v24.1/disaster-recovery-overview.md:169]: nit: i think a comma is needed after 'availability zone'?

**Review by alicia-l2 — COMMENTED** [review]
Added some comments/proposed edits, thanks!

File: src/current/v24.1/disaster-recovery-overview.md [hunk 5]
```diff
@@ -0,0 +1,172 @@
+---
+title: Disaster Recovery Overview
+summary: Learn how to make your cluster resilient with high availability and disaster recovery features.
+toc: true
+---
+
+Resilient deployments aim for continuity in database operation to protect from data loss and down time. To maintain resiliency, it is necessary to build deployments with _high availability_ and _disaster recovery_ coverage.
+
+- [High availability](#choose-a-high-availability-strategy): Continuous and uninterrupted access to data even in the presence of failures or disruptions to maximize uptime.
+- [Disaster recovery](#choose-a-disaster-recovery-strategy): Recovery from a major incident or disaster to minimize downtime and data loss.
```
[the remark this answers is not in the document]
Comment (kathancox) [reply src/current/v24.1/disaster-recovery-overview.md]: Yes, like it. To match the other bullet point, I have changed so they are both verbs.

File: src/current/v24.1/disaster-recovery-overview.md (hunk 1 above)
Comment (kathancox) [reply src/current/v24.1/disaster-recovery-overview.md]: Added, with parenthesis.

File: src/current/v24.1/disaster-recovery-overview.md (hunk 2 above)
Comment (kathancox) [reply src/current/v24.1/disaster-recovery-overview.md]: Added to both columns here.

File: src/current/v24.1/disaster-recovery-overview.md (hunk 3 above)
Comment (kathancox) [reply src/current/v24.1/disaster-recovery-overview.md]: Done!

File: src/current/v24.1/disaster-recovery-overview.md (hunk 4 above)
Comment (kathancox) [reply src/current/v24.1/disaster-recovery-overview.md]: Oh yeah, looks like it! Added the comma!

File: src/current/v24.1/disaster-recovery-overview.md [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+
+{{site.data.alerts.callout_success}}
+For a practical guide on how CockroachDB replicates, distributes, and rebalances data, refer to the [Replication and Rebalancing]({% link {{ page.version.version }}/demo-replication-and-rebalancing.md %}) demo.
+{{site.data.alerts.end}}
+
+## Resilience strategy
+
+As you evaluate CockroachDB's disaster recovery features, consider your organization's requirements for the amount of tolerable data loss and the acceptable length of time to recover.
+
+- Recovery Point Objective (RPO): The maximum amount of data loss (measured by time) that an organization can tolerate losing data.
```
Comment (alicia-l2) [inline src/current/v24.1/disaster-recovery-overview.md:22]: I think "losing data" should be removed?

File: src/current/v24.1/disaster-recovery-overview.md [hunk 7]
```diff
@@ -0,0 +1,172 @@
+---
+title: Disaster Recovery Overview
+summary: Learn how to make your cluster resilient with high availability and disaster recovery features.
+toc: true
+---
+
+Resilient deployments aim for continuity in database operation to protect from data loss and down time. To maintain resiliency, it is necessary to build deployments with _high availability_ and _disaster recovery_ coverage.
```
Comment (alicia-l2) [inline src/current/v24.1/disaster-recovery-overview.md:7]: database operation continuity?

**Review by alicia-l2 — COMMENTED** [review]
Added a few more comments!

File: src/current/v24.1/disaster-recovery-overview.md (hunk 7 above)
Comment (kathancox) [reply src/current/v24.1/disaster-recovery-overview.md:7]: I haven't changed this. Having "database operation" modify "continuity" feels a little harder to read. I have left as-is for now — hopefully my docs review partner may have an idea here.

Comment (alicia-l2) [reply src/current/v24.1/disaster-recovery-overview.md:7]: No worries, sounds good!

**Review by alicia-l2 — APPROVED** [review]
LGTM thanks!

File: src/current/v24.1/disaster-recovery-overview.md (hunk 7 above)
Comment (Amruta-Ranade) [inline src/current/v24.1/disaster-recovery-overview.md:7]:
```suggestion
Resilient deployments aim for continuity in database operations to protect from data loss and downtime. To maintain resiliency, it is necessary to build deployments with _high availability_ and _disaster recovery_ coverage.
```

File: src/current/v24.1/disaster-recovery-overview.md [hunk 8]
```diff
@@ -0,0 +1,172 @@
...
+<img src="{{ 'images/v24.1/rpo-rto.png' | relative_url }}" alt="Simulating RPO and RTO. With RPO representing the tolerable data loss, and RTO representing the tolerable time to recovery." style="border:0px solid #eee;max-width:100%" />
+
+For a comparison of CockroachDB resiliency features, refer to the following sections:
+
+- [Choose a high availability strategy](#choose-a-high-availability-strategy)
+- [Choose a disaster recovery strategy](#choose-a-disaster-recovery-strategy)
+
+### Choose a high availability strategy
+
+CockroachDB uses synchronous, built-in replication to create and distribute copies of data, ensuring consistency across the data copies. The database can tolerate nodes going offline without service interruption whether in a single-region or [multi-region]({% link {{ page.version.version }}/multiregion-overview.md %}) cluster.
```
Comment (Amruta-Ranade) [inline src/current/v24.1/disaster-recovery-overview.md:36]:
```suggestion
CockroachDB uses synchronous, built-in replication to create and distribute copies of data, ensuring consistency across the data copies. The database can tolerate nodes going offline without service interruption, whether in a single-region or [multi-region]({% link {{ page.version.version }}/multiregion-overview.md %}) cluster.
```

File: src/current/v24.1/disaster-recovery-overview.md [hunk 9]
```diff
@@ -0,0 +1,172 @@
...
+    <td class="comparison-chart__feature">
+      <b>Fault tolerance</b>
+    </td>
+    <td>Zero RPO node, availability zone failures</td>
+    <td>Zero RPO node, availability zone failures, region failures</td>
+  </tr>
+
+</table>
+
+For detail on designing your cluster topology for high availability with replication, refer to the [Disaster Recovery Planning]({% link {{ page.version.version }}/disaster-recovery-planning.md %}#hardware-failure) page.
```
Comment (Amruta-Ranade) [inline src/current/v24.1/disaster-recovery-overview.md:103]:
```suggestion
For details on designing your cluster topology for high availability with replication, refer to the [Disaster Recovery Planning]({% link {{ page.version.version }}/disaster-recovery-planning.md %}#hardware-failure) page.
```

File: src/current/v24.1/disaster-recovery-overview.md [hunk 10]
```diff
@@ -0,0 +1,172 @@
...
+
+### Choose a disaster recovery strategy
+
+CockroachDB is designed to recover automatically, however building [backups]({% link {{ page.version.version }}/backup-and-restore-overview.md %}) or [physical cluster replication]({% link {{ page.version.version }}/physical-cluster-replication-overview.md %}) into your disaster recovery planning is an important part of a resilient deployment.
```
Comment (Amruta-Ranade) [inline src/current/v24.1/disaster-recovery-overview.md:107]:
```suggestion
CockroachDB is designed to recover automatically; however, building [backups]({% link {{ page.version.version }}/backup-and-restore-overview.md %}) or [physical cluster replication]({% link {{ page.version.version }}/physical-cluster-replication-overview.md %}) into your disaster recovery planning is an important part of a resilient deployment.
```

**Review by Amruta-Ranade — APPROVED** [review]
A few nits, but overall LGTM!

**Comment by kathancox** [comment]
TFTRs!

**Pull request merged**